proxy/envoy/config.go

```diff
@@ -158,62 +162,88 @@ func buildSidecar(env proxy.Environment, sidecar proxy.Node) (Listeners, Cluster
-
-	listeners := append(inbound, outbound...)
-	clusters := append(inClusters, outClusters...)
-
-	// If management listener port and service port are same, bad things happen
-	// when running in kubernetes, as the probes stop responding. So, append
-	// non overlapping listeners only.
-	for i := range mgmtListeners {
-		m := mgmtListeners[i]
-		c := mgmtClusters[i]
-		l := listeners.GetByAddress(m.Address)
-		if l != nil {
-			glog.Warningf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
-				m.Name, m.Address, l.Name, l.Address)
-			continue
+	listeners := make(Listeners, 0)
+	clusters := make(Clusters, 0)
+
+	if env.Mesh.ProxyListenPort > 0 {
+		inbound, inClusters := buildInboundListeners(env.Mesh, sidecar, instances)
+		outbound, outClusters := buildOutboundListeners(env.Mesh, sidecar, instances, services, env.IstioConfigStore)
+		mgmtListeners, mgmtClusters := buildMgmtPortListeners(env.Mesh, managementPorts, sidecar.IPAddress)
+
+		listeners = append(listeners, inbound...)
+		listeners = append(listeners, outbound...)
+		clusters = append(clusters, inClusters...)
+		clusters = append(clusters, outClusters...)
+
+		// If management listener port and service port are same, bad things happen
+		// when running in kubernetes, as the probes stop responding. So, append
+		// non overlapping listeners only.
+		for i := range mgmtListeners {
+			m := mgmtListeners[i]
+			c := mgmtClusters[i]
+			l := listeners.GetByAddress(m.Address)
+			if l != nil {
+				glog.Warningf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
+					m.Name, m.Address, l.Name, l.Address)
+				continue
+			}
+			listeners = append(listeners, m)
+			clusters = append(clusters, c)
+		}
+
+		// set bind to port values for port redirection
+		for _, listener := range listeners {
+			listener.BindToPort = false
 		}
-		listeners = append(listeners, m)
-		clusters = append(clusters, c)
-	}
 
-	// set bind to port values for port redirection
-	for _, listener := range listeners {
-		listener.BindToPort = false
+		// add an extra listener that binds to the port that is the recipient of the iptables redirect
+		listeners = append(listeners, &Listener{
+			Name:           VirtualListenerName,
+			Address:        fmt.Sprintf("tcp://%s:%d", WildcardAddress, env.Mesh.ProxyListenPort),
+			BindToPort:     true,
+			UseOriginalDst: true,
+			Filters:        make([]*NetworkFilter, 0),
+		})
 	}
 
-	// add an extra listener that binds to the port that is the recipient of the iptables redirect
-	listeners = append(listeners, &Listener{
-		Name:           VirtualListenerName,
-		Address:        fmt.Sprintf("tcp://%s:%d", WildcardAddress, env.Mesh.ProxyListenPort),
-		BindToPort:     true,
-		UseOriginalDst: true,
-		Filters:        make([]*NetworkFilter, 0),
-	})
+	// enable HTTP PROXY port if necessary; this will add an RDS route for this port
+	if env.Mesh.ProxyHttpPort > 0 {
```

Comment:
We had the admin port because Envoy wouldn't start without one, or atleast that was my recollection. Now that we know it's not needed, we should make it optional
with default off. One less listener. 

With regard to the proxy port, we would like to use pilot standalone at lyft. In lieu of that requirement, we need to make sure we can achieve config parity with existing Envoy config and not generate unnecessary listeners (e.g. http proxy). Their system is simple and straightforward - no iptables and no http proxies. They have explicit listeners for ingress and egress path.

I am not really sold on the mixed mode operation with iptables and http proxy on same VM. Can you provide a real world use case for this? 

The endpoint port on sds refers to the port on which a destination VM is listening for inbound traffic. In http proxy mode, we need to specify the destination port as the proxy port on which Envoy is listening or some other dedicated port. This would mean that the vm has to register itself with the Envoy port as its endpoint port. The configuration gets complicated by having some VMs using proxy port and others using iptables.

There is also the issue of tcp services which though is not the subject of this pr, is the primary use case (atleast on ibm end) for integrating VMs with k8s. In the absence of iptables, two VMs talking to one another via tcp will require dedicated port per service- and generating what is otherwise a virtual listener as a physical listener. That would require changes to LDS (the wip pr @kyessenov has for local pilot). All of these become directly applicable in lyft's environment.